packages/fxa-content-server/tests/functional/sign_up.js

```diff
@@ -480,6 +483,23 @@ registerSuite('signup', {
         .then(testElementExists(selectors.SETTINGS.HEADER));
     },
 
+    'signup via product page and redirect after confirm': async function() {
+      return (
+        this.remote
+          .then(openPage(PRODUCT_URL, selectors.SIGNUP.HEADER))
+          .then(fillOutSignUp(email, PASSWORD))
+          .then(testAtConfirmScreen(email))
+          // Note the way getVerificationLink works, it doesn't get the link from the email but instead only gets the
+          // verification code. We have to ask it to add redirectTo=... to recreate the entire URL
+          .then(
+            openVerificationLinkInSameTab(email, 0, {
+              query: { redirectTo: PRODUCT_URL },
```

Comment:
What makes me uncomfortable here is that other tests *do not* set redirectTo, and won't see any change in behavior due to this patch (because the verification link works slightly differently in tests). So I'm not super-comfortable that this isn't regressing anything.

Reply:
It looks like `redirectTo` is used when signing into an account to try to get to panels in the settings page such as account recovery, setting up 2FA, or seeing a list of emails. In BigQuery, I ran this query:

```sql
SELECT * FROM `moz-fx-fxa-prod-0712.fxa_prod_logs.docker_fxa_content_20190726` where jsonPayload.fields.path like '%redirectTo%' LIMIT 100
```

It seems like more explicit tests could be added by opening the signin/signup page with a `redirectTo` of some internal page, maybe the setup 2FA page or something. When the user completes signing in/up, see if the user ends up where we expect them to be.